pkg/internal/owner/owner.go

```diff
@@ -57,8 +57,12 @@ func SetOwnerReference(owner, object runtime.Object, scheme *runtime.Scheme) (ch
 
 	ownerLabels := labelsForOwner(owner, scheme)
 	for k, v := range ownerLabels {
-		if labels[k] != v {
+		if labels[k] == "" {
+			// label was not set before.
 			changed = true
+		} else if labels[k] != v {
+			// label is overriden.
+			return changed, fmt.Errorf("tried to override owner reference: %s=%s to =%s", k, labels[k], v)
 		}
 		labels[k] = v
```

Comment:
this could be moved in the `if labels[k] == ""` section since it's only setted there.